### ISSUE TITLE
QueryCountStatsMonitor doesn't track sql query metrics

### ISSUE THREAD

**Issue by a2l007** [apache/druid#11652]
QueryCountStatsMonitor provides the following query metrics
`query/count` 
`query/success/count`
`query/failed/count`
`query/interrupted/count`
 `query/timeout/count`
It looks like these metrics don't track the sql queries invoked against the `druid/v2/sql` endpoint.
This could be because the Broker is tied to `BrokerQueryResource` as the `QueryCountStatsProvider` implementation and even if `SqlResource` implements `QueryCountStatsProvider`, it wouldn't work. 
Unless the SqlResource was designed as such for some other reason, I'm looking at merging the `SqlResource` into `QueryResource` so that `druid/v2/sql` can take advantage of the query count stats monitoring within `QueryResource`.

@jihoonson Any thoughts on this?

**Comment by maytasm** [comment]
@a2l007 Did you have a chance to work on this?

**Comment by a2l007** [comment]
@maytasm Nope, but looks like there is already a PR out for this!

**Comment by maytasm** [comment]
@a2l007 Yep. If you have a few minutes, would appreciate if you can review the PR https://github.com/apache/druid/pull/17452 Thanks!
